Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -1350,12 +1389,13 @@ static gboolean handleScroll(WebKitWebViewBase* webViewBase, double deltaX, doub
         // We only have hold events on touchpads, so assume one if we get no event.
         GdkInputSource source = device ? gdk_device_get_source(device) : GDK_SOURCE_TOUCHPAD;
 
+        bool isEnd = event ? gdk_event_is_scroll_stop_event(event) : false;
         PlatformGtkScrollData scrollData = { .delta = delta, .eventTime = eventTime, .source = source, .isEnd = isEnd };
         if (controller->handleScrollWheelEvent(&scrollData))
             return GDK_EVENT_STOP;
     }
 
-    if (!isEnd && shouldInvertDirectionForScrollEvent(priv->mouseIsOverScrollbar, gdk_event_get_modifier_state(event) & GDK_SHIFT_MASK))
+    if (shouldInvertDirectionForScrollEvent(priv->mouseIsOverScrollbar, gdk_event_get_modifier_state(event) & GDK_SHIFT_MASK))
```

Comment:
This use? Well, one thing with null events is that can only happen for scroll-end (they are synthesized if you have old libinput without hold gestures support/on trackpoint/possible on semi-mt touchpads, dunno about those.

So the !isEnd check here was accounting for that. :)

Reply:
Sure, I added the !isEnd check back here. I don't know these libinput edge cases, so feel free to be prescriptive about how to handle them. e.g. do I need to bring back the scroll-end listener for this?

Reply:
Done